bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -797,7 +849,7 @@ public int getReadOffset()
      * @return boolean
      */
     @Override
-    public synchronized boolean hasNext()
```

Comment:
For my understanding, why was this method synchronized in first place itself? If I understand correctly, isnt this method local to a single Subscriber instance?